src/Middleware/OutputCaching/src/Global.cs

```diff
@@ -0,0 +1,6 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+[module: SkipLocalsInit] // all stackalloc usage checked
```

Comment:
Given C#'s definite assignment, there's only 3 main ways of doing that:

- stackalloc
- calling `out` methods that ignore the semantics (such as `Unsafe.SkipInit')
- intentional pointer math

The last two are fully self-inflicted. The performance benefits of skipping locals init can apply generally, big are observed in particular in stackalloc. But: huge chunks of the runtime now use this mode by default. I genuinely don't think we should shy away from it.

However! I'm OK with compromise - how would you feel about moving the attribute to the method where I'm using the stackalloc for now, and we can look at the rest later on a separate ticket?

I can also run the perf suite in both modes, so we can see the impact in concrete terms.

Reply:
> However! I'm OK with compromise - how would you feel about moving the attribute to the method where I'm using the stackalloc for now, and we can look at the rest later on a separate ticket?

That's fine. I had a PR that enabled SkipLocalsInit across many projects. Other people were gun-shy about globally enabling it so it was reduced to methods where stackalloc is used.

Reply:
https://github.com/search?q=repo%3Adotnet%2Faspnetcore%20SkipLocalsInit&type=code

Reply:
I genuinely think we can be more bold about it; IIRC it is on by default for "runtime" ([citation](https://github.com/dotnet/runtime/blob/408729ca215ef5e9ff5127d878b042c4ff3849e9/src/libraries/Directory.Build.targets#L200)), and I could only find a single project that opts out (although there is definitely some validation work per project on the edge cases I mentioned). However, I'm happy to take that as a separate follow up item where we can focus on it as a single dedicated thing, with profiling etc; maybe talk to a few folks like @stephentoub / @jaredpar. For now I'll move it local to just the stackalloc usage.

Reply:
I made the same points. Closed PR with discussion: https://github.com/dotnet/aspnetcore/pull/26598

I didn't feel strongly enough about it to keep pushing, but you're welcome to develop a good argument 😄 A notable difference in RPS is the best evidence.

Reply:
Useful context. I'm happy to take this as a personal mission, but: I don't need to do that in this PR. I will happily eat the work to get the evidence :)

Reply:
> It's been 3 years since runtime enabled skip locals init

FWIW, it's been more than that. We switched to using SkipLocalsInit when it was a supported language feature, but before that we achieved the same thing via a linker plugin that stripped out the locals init. I believe we enabled that in NET Core 2.1.